### PR TITLE
Maciej grochowski

### PR DESCRIPTION
**Pull request by MaciejGrochowski93**
I have prepared some API methods, which are supposed to allow users to find/add/remove meals from the main & favorite lists. Please, take a look.

### CODE REVIEW

**Review by LachowskiTomekPL — APPROVED** [review]

File: src/main/java/pl/console/project/model/ListMeal.java [hunk 1]
```diff
@@ -18,8 +19,53 @@ public ListMeal(@JsonProperty("meals") List<Meal> listMeals) {
         return listMeals;
     }
 
+    public Meal findMealByName(String name) {
+        for (Meal meal : listMeals) {
+            if (meal.getName().equals(name)) {
+                return meal;
+            }
+        }
+        return null;
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/model/ListMeal.java]: return null is not good programming practice, think about Optional<Meal>

File: src/main/java/pl/console/project/model/ListMeal.java [hunk 2]
```diff
@@ -18,8 +19,53 @@ public ListMeal(@JsonProperty("meals") List<Meal> listMeals) {
...
+        List<Meal> mealsCategoryList = new LinkedList<>();
+        for (Meal meal : listMeals) {
+            if (meal.getCategory().equals(category)) {
+                mealsCategoryList.add(meal);
+            }
+        }
+        return mealsCategoryList;
+    }
+
+    public List<Meal> findMealByIngredient(String ingredient) { // shift+f6
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/model/ListMeal.java:48]: // shift+f6 please remove comment before PR

File: src/main/java/pl/console/project/model/ListMeal.java [hunk 3]
```diff
@@ -18,8 +19,53 @@ public ListMeal(@JsonProperty("meals") List<Meal> listMeals) {
...
+    }
+
+    public void removeMealFromList(Meal meal) {
+        for (Meal existingMeal : listMeals){
+            if (existingMeal.getName().equals(meal.getName())){
+                listMeals.remove(meal);
+            }
+        }
+    }
+
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/model/ListMeal.java:73]: Please use often ctr+alt+l ;)

File: src/main/java/pl/console/project/repositories/Repository.java [hunk 4]
```diff
@@ -27,6 +32,53 @@ public ListMeal getListMeal() {
         return listMeal;
     }
 
+//<<<<<<< HEAD
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/repositories/Repository.java]: Something went wrong when You merged, please remove this line

File: src/main/java/pl/console/project/repositories/Repository.java [hunk 5]
```diff
@@ -27,6 +32,53 @@ public ListMeal getListMeal() {
...
+    }
+
+    public void addMealToList(Meal meal) {
+        listMeal.addMealToList(meal);
+    }
+
+    public void removeMealFromList(Meal meal) {
+        listMeal.removeMealFromList(meal);
+    }
+//=======
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/repositories/Repository.java]: Something went wrong when You merged, please remove this line

File: src/main/java/pl/console/project/repositories/Repository.java [hunk 6]
```diff
@@ -27,6 +32,53 @@ public ListMeal getListMeal() {
...
+    }
+
+    public void addMealToFavorites(Meal meal){
+        favoriteListMeal.addMealToList(meal);
+    }
+
+    public void removeMealFromFavorites(Meal meal){
+        favoriteListMeal.removeMealFromList(meal);
+    }
+//>>>>>>> origin/MarcinKozak
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/repositories/Repository.java:80]: Something went wrong when You merged, please remove this line

File: src/main/java/pl/console/project/repositories/Repository.java [hunk 7]
```diff
@@ -39,4 +91,16 @@ static ListMeal readFile(String path) {
...
+        ObjectMapper objectMapper = new ObjectMapper();
+        ListMeal favoriteListMeal = null;
+
+        try {
+            objectMapper.writeValue(new File("src/main/resources/favorite meal list.json"), favoriteListMeal);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return favoriteListMeal;
+    }
```
Comment (RozwalkaWojciech) [inline src/main/java/pl/console/project/repositories/Repository.java:105]: Remove this method because, MarcinKozak should rewrite it ;)

**Review by RozwalkaWojciech — COMMENTED** [review]
Please clean your code

**Pull request merged**